core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -123,6 +126,10 @@ private DefaultClientRequestContext(DefaultClientRequestContext ctx, Request req
         responseTimeoutMillis = ctx.responseTimeoutMillis();
         maxResponseLength = ctx.maxResponseLength();
 
+        final HttpHeaders additionalHeaders = ctx.additionalRequestHeaders();
+        additionalRequestHeaders = new DefaultHttpHeaders(true, additionalHeaders.size());
+        additionalRequestHeaders.set(additionalHeaders);
```

Comment:
Ditto - We don't really need to make a copy or to specify the initial capacity if the source headers is empty.